src/vs/workbench/parts/extensions/electron-browser/extensionsViews.ts

```diff
@@ -801,7 +801,7 @@ export class WorkspaceRecommendedExtensionsView extends ExtensionsListView {
 	}
 
 	async show(query: string): Promise<IPagedModel<IExtension>> {
-		let model = await ((query && query.trim() !== '@recommended') ? this.showEmptyModel() : super.show(this.recommendedExtensionsQuery));
+		let model = await (/@recommended/.test(query) ? super.show(this.recommendedExtensionsQuery) : this.showEmptyModel());
```

Comment:
In the scenario which we are trying to fix, what is the value of `query`?

Reply:
`@recommened:workspace`

Reply:
Then the right fix should be that we should check with `this.recommendedExtensionsQuery` instead of `@recommended`

Reply:
We wouldn't want the `WorkspaceRecommendedExtensionsView` to run its query in case of say `@recommended Go`. So the right fix here would be `query && query.trim() !== '@recommended' && query.trim() !== '@recommended:workspace'